src/web/mjs/engine/Storage.mjs

```diff
@@ -430,9 +430,10 @@ export default class Storage {
      */
     saveChapterPages( chapter, content ) {
         try {
+            let leadingZeroes = '0'.repeat( String(content.length).length - 1 );
             let pageData = content.map( ( page, index ) => {
                 return {
-                    name: this._pageFileName( index + 1, page.type ),
+                    name: this._pageFileName( index + 1, page.type, leadingZeroes ),
```

Comment:
Consider providing the target length as number instead as string and let `_pageFileName` do the `'0'` padding